### PR TITLE
Blazor API Review: Compiler Services Part 2

### PR DESCRIPTION
**Pull request by rynowak**
Part of: #11907

### CODE REVIEW

File: src/Components/Components/src/RuntimeHelpers.cs [hunk 1]
```diff
@@ -17,6 +18,43 @@ public static class RuntimeHelpers
...
+        /// <param name="receiver"></param>
+        /// <param name="callback"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        //
+        // This method is used with `@bind-Value` for components. When a component has a generic type, it's
+        // really messy to write to try and write the parameter type for ValueChanged - because it can contain generic
+        // type parameters. We're using a trick of type inference to generate the proper typing for the delegate
+        // so that method-group-to-delegate conversion works.
+        public static EventCallback<T> CreateInferredEventCallback<T>(object receiver, Action<T> callback, T value)
```
Comment (rynowak) [inline src/Components/Components/src/RuntimeHelpers.cs]: Need to add a new method first, then we can remove the old one after the compiler updates.

**Comment by rynowak** [comment]
@SteveSandersonMS

**Review by SteveSandersonMS — APPROVED** [review]
Looks like you're doing this to enable some compiler work, after which you'll remove the original methods. Please let me know if I'm wrong :)

**Comment by rynowak** [comment]
> Looks like you're doing this to enable some compiler work, after which you'll remove the original methods. Please let me know if I'm wrong :)

Exactly right.

**Pull request merged**